loopd/run.go

```diff
@@ -168,7 +170,27 @@ func Start(rpcCfg RPCConfig) error {
 
 	// Execute command.
 	if parser.Active == nil {
-		return daemon(&config, lisCfg)
+		signal.Intercept()
+
+		errChan := make(chan error)
+		daemon := New(&config, lisCfg, errChan)
+		if err := daemon.Start(); err != nil {
+			return err
+		}
+
+		select {
+		case <-signal.ShutdownChannel():
+			log.Infof("Received SIGINT (Ctrl+C).")
+			return daemon.Stop()
+
+		case err := <-errChan:
+			log.Errorf("Daemon sent unexpected error: %v", err)
+			stopErr := daemon.Stop()
```

Comment:
I would think that this shouldn't be necessary. The daemon should have shut down cleanly if it sends the error, to keep responsibilities clean.